presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -307,7 +307,12 @@ protected Type visitDereferenceExpression(DereferenceExpression node, StackableA
                     return field.getType();
                 }
 
-                assertColumnPrefix(qualifiedName, node);
+                if (!isColumnPrefix(qualifiedName, tupleDescriptor)) {
+                    if (isReferenceToOuterRelation(context.getContext(), qualifiedName)) {
+                        throw new SemanticException(NOT_SUPPORTED, node, "Accessing outer relation symbol '%s', while correlated subqueries are not yet supported", node);
```

Comment:
The notion of "symbol" is an internal detail of the analyzer/planner. It'd be better to say "column" or "field". I would rephrase this as:

"Correlated subqueries are not yet supported. Invalid column reference: %s"